runtime/jcl/common/thread.cpp

```diff
@@ -589,6 +589,44 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+static UDATA findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->userData1) {
+		rc = J9_STACKWALK_STOP_ITERATING;
+	} else if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		char *methodName = (char *)J9UTF8_DATA(J9ROMMETHOD_NAME(romMethod));
+
+		if (((J9VMJAVALANGTHREAD(vm) == methodClass)
+			|| (J9VMJAVALANGVIRTUALTHREAD(vm) == methodClass)
+			|| (J9VMJDKINCUBATORCONCURRENTSCOPEDVALUECARRIER(vm) == methodClass))
```

Comment:
Does the vmconstantpool macro ensure `_OR_NULL` will print to the class when vm loaded it? I recall there was some issue with those limited to what is in the requiredClasses list

Reply:
The `_OR_NULL` versions simply fetch the result from the constant pool with no resolution. They are mainly used for classes we know to have been pre-populated in the constant pool, i.e. any class in the constant pool that has fields or methods. In those cases, we don't check for the NULL result at all, assuming the slot is valid.

Reply:
Fine for `Thread` and `VirtualThread` but not for `jdk/incubator/concurrent/ScopedValue$Carrier` since it declares no fields or method in the constant pool. Options are - declare a field, preload it manually during startup or handle the unresolved case here.

Reply:
I did encountered the case where I'm looking for an instance of `jdk/incubator/concurrent/ScopedValue$Carrier` but the `OR_NULL` version returns null.